File: 2015/go/src/day1A.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func sumValArray(valArray []int) int {
	total := 0

	for _, val := range valArray {
		total = total + val
	}

	return total
}

func convertValArrayToInt(valArray []string) []int {
	var intArray []int

	for _, val := range valArray {
		intVal, err := strconv.Atoi(val)
		check(err)
		intArray = append(intArray, intVal)
	}

	return intArray
}

func main() {
	input := readFile("day1.dat")

	//Replace Strings with numeric values
	input = strings.TrimSpace(strings.ReplaceAll(strings.ReplaceAll(input, "(", "1 "), ")", "-1 "))
	valArray := strings.Split(input, " ")

	total := sumValArray(convertValArrayToInt(valArray))

	fmt.Println(total)
}

File: 2020/python/day1A.py
import common

def main():
    dat = common.readFile('day1.dat')
    
    val = find_pair(dat)
    print(val[0]*val[1])


def find_pair(input):
    found = False
    i = 0
    val = (0, 0)

    while (not found and i < len(input)):
        j = i + 1
        while (not found and j < len(input)):
            found = (int(input[i]) + int(input[j])) == 2020
            if found:
                val = (int(input[i]), int(input[j]))
                
            
            j = j + 1
        
        i = i + 1
    
    return val


if __name__ == '__main__':
    main()